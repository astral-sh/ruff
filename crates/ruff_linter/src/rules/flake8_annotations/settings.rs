//! Settings for the `flake-annotations` plugin.

use crate::display_settings;
use ruff_macros::CacheKey;
use std::fmt::{Display, Formatter};

#[derive(Debug, Default, CacheKey)]
#[allow(clippy::struct_excessive_bools)]
pub struct Settings {
    pub mypy_init_return: bool,
    pub suppress_dummy_args: bool,
    pub suppress_none_returning: bool,
    pub allow_star_arg_any: bool,
    pub ignore_fully_untyped: bool,
}

impl Display for Settings {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        display_settings! {
            formatter = f,
            namespace = "linter.flake8_annotations",
            fields = [
                self.mypy_init_return,
                self.suppress_dummy_args,
                self.suppress_none_returning,
                self.allow_star_arg_any,
                self.ignore_fully_untyped
            ]
        }
        Ok(())
    }
}
