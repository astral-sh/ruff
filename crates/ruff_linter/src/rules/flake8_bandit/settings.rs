//! Settings for the `flake8-bandit` plugin.

use ruff_macros::CacheKey;

pub fn default_tmp_dirs() -> Vec<String> {
    ["/tmp", "/var/tmp", "/dev/shm"]
        .map(ToString::to_string)
        .to_vec()
}

#[derive(Debug, CacheKey)]
pub struct Settings {
    pub hardcoded_tmp_directory: Vec<String>,
    pub check_typed_exception: bool,
}

impl Default for Settings {
    fn default() -> Self {
        Self {
            hardcoded_tmp_directory: default_tmp_dirs(),
            check_typed_exception: false,
        }
    }
}
