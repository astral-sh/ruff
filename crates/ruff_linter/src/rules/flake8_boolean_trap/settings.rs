//! Settings for the `flake8-boolean-trap` plugin.

use std::fmt;

use ruff_macros::CacheKey;

use crate::display_settings;

#[derive(Debug, Clone, CacheKey, Default)]
pub struct Settings {
    pub extend_allowed_calls: Vec<String>,
}

impl fmt::Display for Settings {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        display_settings! {
            formatter = f,
            namespace = "linter.flake8_boolean_trap",
            fields = [
                self.extend_allowed_calls | array,
            ]
        }
        Ok(())
    }
}
