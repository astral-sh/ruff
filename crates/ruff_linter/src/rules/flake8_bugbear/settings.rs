//! Settings for the `flake8-bugbear` plugin.

use ruff_macros::CacheKey;

#[derive(Debug, Default, CacheKey)]
pub struct Settings {
    pub extend_immutable_calls: Vec<String>,
}
