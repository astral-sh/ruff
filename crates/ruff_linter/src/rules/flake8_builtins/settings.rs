//! Settings for the `flake8-builtins` plugin.

use crate::display_settings;
use ruff_macros::CacheKey;
use std::fmt::{Display, Formatter};

#[derive(Debug, Default, CacheKey)]
pub struct Settings {
    pub builtins_ignorelist: Vec<String>,
}

impl Display for Settings {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        display_settings! {
            formatter = f,
            namespace = "linter.flake8_builtins",
            fields = [
                self.builtins_ignorelist | array
            ]
        }
        Ok(())
    }
}
