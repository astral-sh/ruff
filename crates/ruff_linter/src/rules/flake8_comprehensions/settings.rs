//! Settings for the `flake8-comprehensions` plugin.

use ruff_macros::CacheKey;

#[derive(Debug, Default, CacheKey)]
pub struct Settings {
    pub allow_dict_calls_with_keyword_arguments: bool,
}
