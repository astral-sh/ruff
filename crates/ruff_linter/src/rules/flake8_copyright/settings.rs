//! Settings for the `flake8-copyright` plugin.

use once_cell::sync::Lazy;
use regex::Regex;
use std::fmt::{Display, Formatter};

use crate::display_settings;
use ruff_macros::CacheKey;

#[derive(Debug, CacheKey)]
pub struct Settings {
    pub notice_rgx: Regex,
    pub author: Option<String>,
    pub min_file_size: usize,
}

pub static COPYRIGHT: Lazy<Regex> =
    Lazy::new(|| Regex::new(r"(?i)Copyright\s+(\(C\)\s+)?\d{4}(-\d{4})*").unwrap());

impl Default for Settings {
    fn default() -> Self {
        Self {
            notice_rgx: COPYRIGHT.clone(),
            author: None,
            min_file_size: 0,
        }
    }
}

impl Display for Settings {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        display_settings! {
            formatter = f,
            namespace = "linter.flake8_copyright",
            fields = [
                self.notice_rgx,
                self.author | optional,
                self.min_file_size,
            ]
        }
        Ok(())
    }
}
