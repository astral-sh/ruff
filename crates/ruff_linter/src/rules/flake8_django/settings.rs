//! Settings for the `flake8-django` plugin.

use crate::display_settings;
use ruff_macros::CacheKey;
use std::fmt::{Display, Formatter};

#[derive(Debug, Clone, CacheKey)]
pub struct Settings {
    pub additional_path_functions: Vec<String>,
}

impl Default for Settings {
    fn default() -> Self {
        Self {
            additional_path_functions: vec![],
        }
    }
}

impl Display for Settings {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        display_settings! {
            formatter = f,
            namespace = "linter.flake8_django",
            fields = [
                self.additional_path_functions | array
            ]
        }
        Ok(())
    }
}
