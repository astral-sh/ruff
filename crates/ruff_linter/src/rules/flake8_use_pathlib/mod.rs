//! Rules from [flake8-use-pathlib](https://pypi.org/project/flake8-use-pathlib/).
pub(crate) mod rules;
pub(crate) mod violations;

#[cfg(test)]
mod tests {
    use std::path::Path;

    use anyhow::Result;
    use test_case::test_case;

    use crate::assert_messages;
    use crate::registry::Rule;
    use crate::settings;
    use crate::test::test_path;

    #[test_case(Path::new("full_name.py"))]
    #[test_case(Path::new("import_as.py"))]
    #[test_case(Path::new("import_from_as.py"))]
    #[test_case(Path::new("import_from.py"))]
    #[test_case(Path::new("use_pathlib.py"))]
    fn rules(path: &Path) -> Result<()> {
        let snapshot = format!("{}", path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("flake8_use_pathlib").join(path).as_path(),
            &settings::LinterSettings::for_rules(vec![
                Rule::OsPathAbspath,
                Rule::OsChmod,
                Rule::OsMkdir,
                Rule::OsMakedirs,
                Rule::OsRename,
                Rule::OsReplace,
                Rule::OsRmdir,
                Rule::OsRemove,
                Rule::OsUnlink,
                Rule::OsGetcwd,
                Rule::OsPathExists,
                Rule::OsPathExpanduser,
                Rule::OsPathIsdir,
                Rule::OsPathIsfile,
                Rule::OsPathIslink,
                Rule::OsReadlink,
                Rule::OsStat,
                Rule::OsPathIsabs,
                Rule::OsPathJoin,
                Rule::OsPathBasename,
                Rule::OsPathDirname,
                Rule::OsPathSamefile,
                Rule::OsPathSplitext,
                Rule::BuiltinOpen,
            ]),
        )?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }

    #[test_case(Rule::PyPath, Path::new("py_path_1.py"))]
    #[test_case(Rule::PyPath, Path::new("py_path_2.py"))]
    #[test_case(Rule::PathConstructorCurrentDirectory, Path::new("PTH201.py"))]
    #[test_case(Rule::OsPathGetsize, Path::new("PTH202.py"))]
    #[test_case(Rule::OsPathGetatime, Path::new("PTH203.py"))]
    #[test_case(Rule::OsPathGetmtime, Path::new("PTH204.py"))]
    #[test_case(Rule::OsPathGetctime, Path::new("PTH205.py"))]
    #[test_case(Rule::OsSepSplit, Path::new("PTH206.py"))]
    #[test_case(Rule::Glob, Path::new("PTH207.py"))]
    fn rules_pypath(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.noqa_code(), path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("flake8_use_pathlib").join(path).as_path(),
            &settings::LinterSettings::for_rule(rule_code),
        )?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }
}
