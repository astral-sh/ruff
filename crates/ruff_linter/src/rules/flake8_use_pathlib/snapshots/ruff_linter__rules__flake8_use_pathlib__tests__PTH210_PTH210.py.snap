---
source: crates/ruff_linter/src/rules/flake8_use_pathlib/mod.rs
snapshot_kind: text
---
PTH210.py:21:1: PTH210 Invalid suffix passed to `.with_suffix()`
   |
20 | ### Errors
21 | path.with_suffix(".")
   | ^^^^^^^^^^^^^^^^^^^^^ PTH210
22 | path.with_suffix("py")
23 | path.with_suffix(r"s")
   |
   = help: Remove "." or extend to valid suffix

PTH210.py:22:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
20 | ### Errors
21 | path.with_suffix(".")
22 | path.with_suffix("py")
   | ^^^^^^^^^^^^^^^^^^^^^^ PTH210
23 | path.with_suffix(r"s")
24 | path.with_suffix(u'' "json")
   |
   = help: Add a leading dot

ℹ Unsafe fix
19 19 | 
20 20 | ### Errors
21 21 | path.with_suffix(".")
22    |-path.with_suffix("py")
   22 |+path.with_suffix(".py")
23 23 | path.with_suffix(r"s")
24 24 | path.with_suffix(u'' "json")
25 25 | path.with_suffix(suffix="js")

PTH210.py:23:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
21 | path.with_suffix(".")
22 | path.with_suffix("py")
23 | path.with_suffix(r"s")
   | ^^^^^^^^^^^^^^^^^^^^^^ PTH210
24 | path.with_suffix(u'' "json")
25 | path.with_suffix(suffix="js")
   |
   = help: Add a leading dot

ℹ Unsafe fix
20 20 | ### Errors
21 21 | path.with_suffix(".")
22 22 | path.with_suffix("py")
23    |-path.with_suffix(r"s")
   23 |+path.with_suffix(r".s")
24 24 | path.with_suffix(u'' "json")
25 25 | path.with_suffix(suffix="js")
26 26 | 

PTH210.py:24:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
22 | path.with_suffix("py")
23 | path.with_suffix(r"s")
24 | path.with_suffix(u'' "json")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
25 | path.with_suffix(suffix="js")
   |
   = help: Add a leading dot

ℹ Unsafe fix
21 21 | path.with_suffix(".")
22 22 | path.with_suffix("py")
23 23 | path.with_suffix(r"s")
24    |-path.with_suffix(u'' "json")
   24 |+path.with_suffix(u'.' "json")
25 25 | path.with_suffix(suffix="js")
26 26 | 
27 27 | posix_path.with_suffix(".")

PTH210.py:25:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
23 | path.with_suffix(r"s")
24 | path.with_suffix(u'' "json")
25 | path.with_suffix(suffix="js")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
26 | 
27 | posix_path.with_suffix(".")
   |
   = help: Add a leading dot

ℹ Unsafe fix
22 22 | path.with_suffix("py")
23 23 | path.with_suffix(r"s")
24 24 | path.with_suffix(u'' "json")
25    |-path.with_suffix(suffix="js")
   25 |+path.with_suffix(suffix=".js")
26 26 | 
27 27 | posix_path.with_suffix(".")
28 28 | posix_path.with_suffix("py")

PTH210.py:27:1: PTH210 Invalid suffix passed to `.with_suffix()`
   |
25 | path.with_suffix(suffix="js")
26 | 
27 | posix_path.with_suffix(".")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
28 | posix_path.with_suffix("py")
29 | posix_path.with_suffix(r"s")
   |
   = help: Remove "." or extend to valid suffix

PTH210.py:28:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
27 | posix_path.with_suffix(".")
28 | posix_path.with_suffix("py")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
29 | posix_path.with_suffix(r"s")
30 | posix_path.with_suffix(u'' "json")
   |
   = help: Add a leading dot

ℹ Unsafe fix
25 25 | path.with_suffix(suffix="js")
26 26 | 
27 27 | posix_path.with_suffix(".")
28    |-posix_path.with_suffix("py")
   28 |+posix_path.with_suffix(".py")
29 29 | posix_path.with_suffix(r"s")
30 30 | posix_path.with_suffix(u'' "json")
31 31 | posix_path.with_suffix(suffix="js")

PTH210.py:29:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
27 | posix_path.with_suffix(".")
28 | posix_path.with_suffix("py")
29 | posix_path.with_suffix(r"s")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
30 | posix_path.with_suffix(u'' "json")
31 | posix_path.with_suffix(suffix="js")
   |
   = help: Add a leading dot

ℹ Unsafe fix
26 26 | 
27 27 | posix_path.with_suffix(".")
28 28 | posix_path.with_suffix("py")
29    |-posix_path.with_suffix(r"s")
   29 |+posix_path.with_suffix(r".s")
30 30 | posix_path.with_suffix(u'' "json")
31 31 | posix_path.with_suffix(suffix="js")
32 32 | 

PTH210.py:30:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
28 | posix_path.with_suffix("py")
29 | posix_path.with_suffix(r"s")
30 | posix_path.with_suffix(u'' "json")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
31 | posix_path.with_suffix(suffix="js")
   |
   = help: Add a leading dot

ℹ Unsafe fix
27 27 | posix_path.with_suffix(".")
28 28 | posix_path.with_suffix("py")
29 29 | posix_path.with_suffix(r"s")
30    |-posix_path.with_suffix(u'' "json")
   30 |+posix_path.with_suffix(u'.' "json")
31 31 | posix_path.with_suffix(suffix="js")
32 32 | 
33 33 | pure_path.with_suffix(".")

PTH210.py:31:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
29 | posix_path.with_suffix(r"s")
30 | posix_path.with_suffix(u'' "json")
31 | posix_path.with_suffix(suffix="js")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
32 | 
33 | pure_path.with_suffix(".")
   |
   = help: Add a leading dot

ℹ Unsafe fix
28 28 | posix_path.with_suffix("py")
29 29 | posix_path.with_suffix(r"s")
30 30 | posix_path.with_suffix(u'' "json")
31    |-posix_path.with_suffix(suffix="js")
   31 |+posix_path.with_suffix(suffix=".js")
32 32 | 
33 33 | pure_path.with_suffix(".")
34 34 | pure_path.with_suffix("py")

PTH210.py:33:1: PTH210 Invalid suffix passed to `.with_suffix()`
   |
31 | posix_path.with_suffix(suffix="js")
32 | 
33 | pure_path.with_suffix(".")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
34 | pure_path.with_suffix("py")
35 | pure_path.with_suffix(r"s")
   |
   = help: Remove "." or extend to valid suffix

PTH210.py:34:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
33 | pure_path.with_suffix(".")
34 | pure_path.with_suffix("py")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
35 | pure_path.with_suffix(r"s")
36 | pure_path.with_suffix(u'' "json")
   |
   = help: Add a leading dot

ℹ Unsafe fix
31 31 | posix_path.with_suffix(suffix="js")
32 32 | 
33 33 | pure_path.with_suffix(".")
34    |-pure_path.with_suffix("py")
   34 |+pure_path.with_suffix(".py")
35 35 | pure_path.with_suffix(r"s")
36 36 | pure_path.with_suffix(u'' "json")
37 37 | pure_path.with_suffix(suffix="js")

PTH210.py:35:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
33 | pure_path.with_suffix(".")
34 | pure_path.with_suffix("py")
35 | pure_path.with_suffix(r"s")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
36 | pure_path.with_suffix(u'' "json")
37 | pure_path.with_suffix(suffix="js")
   |
   = help: Add a leading dot

ℹ Unsafe fix
32 32 | 
33 33 | pure_path.with_suffix(".")
34 34 | pure_path.with_suffix("py")
35    |-pure_path.with_suffix(r"s")
   35 |+pure_path.with_suffix(r".s")
36 36 | pure_path.with_suffix(u'' "json")
37 37 | pure_path.with_suffix(suffix="js")
38 38 | 

PTH210.py:36:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
34 | pure_path.with_suffix("py")
35 | pure_path.with_suffix(r"s")
36 | pure_path.with_suffix(u'' "json")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
37 | pure_path.with_suffix(suffix="js")
   |
   = help: Add a leading dot

ℹ Unsafe fix
33 33 | pure_path.with_suffix(".")
34 34 | pure_path.with_suffix("py")
35 35 | pure_path.with_suffix(r"s")
36    |-pure_path.with_suffix(u'' "json")
   36 |+pure_path.with_suffix(u'.' "json")
37 37 | pure_path.with_suffix(suffix="js")
38 38 | 
39 39 | pure_posix_path.with_suffix(".")

PTH210.py:37:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
35 | pure_path.with_suffix(r"s")
36 | pure_path.with_suffix(u'' "json")
37 | pure_path.with_suffix(suffix="js")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
38 | 
39 | pure_posix_path.with_suffix(".")
   |
   = help: Add a leading dot

ℹ Unsafe fix
34 34 | pure_path.with_suffix("py")
35 35 | pure_path.with_suffix(r"s")
36 36 | pure_path.with_suffix(u'' "json")
37    |-pure_path.with_suffix(suffix="js")
   37 |+pure_path.with_suffix(suffix=".js")
38 38 | 
39 39 | pure_posix_path.with_suffix(".")
40 40 | pure_posix_path.with_suffix("py")

PTH210.py:39:1: PTH210 Invalid suffix passed to `.with_suffix()`
   |
37 | pure_path.with_suffix(suffix="js")
38 | 
39 | pure_posix_path.with_suffix(".")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
40 | pure_posix_path.with_suffix("py")
41 | pure_posix_path.with_suffix(r"s")
   |
   = help: Remove "." or extend to valid suffix

PTH210.py:40:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
39 | pure_posix_path.with_suffix(".")
40 | pure_posix_path.with_suffix("py")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
41 | pure_posix_path.with_suffix(r"s")
42 | pure_posix_path.with_suffix(u'' "json")
   |
   = help: Add a leading dot

ℹ Unsafe fix
37 37 | pure_path.with_suffix(suffix="js")
38 38 | 
39 39 | pure_posix_path.with_suffix(".")
40    |-pure_posix_path.with_suffix("py")
   40 |+pure_posix_path.with_suffix(".py")
41 41 | pure_posix_path.with_suffix(r"s")
42 42 | pure_posix_path.with_suffix(u'' "json")
43 43 | pure_posix_path.with_suffix(suffix="js")

PTH210.py:41:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
39 | pure_posix_path.with_suffix(".")
40 | pure_posix_path.with_suffix("py")
41 | pure_posix_path.with_suffix(r"s")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
42 | pure_posix_path.with_suffix(u'' "json")
43 | pure_posix_path.with_suffix(suffix="js")
   |
   = help: Add a leading dot

ℹ Unsafe fix
38 38 | 
39 39 | pure_posix_path.with_suffix(".")
40 40 | pure_posix_path.with_suffix("py")
41    |-pure_posix_path.with_suffix(r"s")
   41 |+pure_posix_path.with_suffix(r".s")
42 42 | pure_posix_path.with_suffix(u'' "json")
43 43 | pure_posix_path.with_suffix(suffix="js")
44 44 | 

PTH210.py:42:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
40 | pure_posix_path.with_suffix("py")
41 | pure_posix_path.with_suffix(r"s")
42 | pure_posix_path.with_suffix(u'' "json")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
43 | pure_posix_path.with_suffix(suffix="js")
   |
   = help: Add a leading dot

ℹ Unsafe fix
39 39 | pure_posix_path.with_suffix(".")
40 40 | pure_posix_path.with_suffix("py")
41 41 | pure_posix_path.with_suffix(r"s")
42    |-pure_posix_path.with_suffix(u'' "json")
   42 |+pure_posix_path.with_suffix(u'.' "json")
43 43 | pure_posix_path.with_suffix(suffix="js")
44 44 | 
45 45 | pure_windows_path.with_suffix(".")

PTH210.py:43:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
41 | pure_posix_path.with_suffix(r"s")
42 | pure_posix_path.with_suffix(u'' "json")
43 | pure_posix_path.with_suffix(suffix="js")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
44 | 
45 | pure_windows_path.with_suffix(".")
   |
   = help: Add a leading dot

ℹ Unsafe fix
40 40 | pure_posix_path.with_suffix("py")
41 41 | pure_posix_path.with_suffix(r"s")
42 42 | pure_posix_path.with_suffix(u'' "json")
43    |-pure_posix_path.with_suffix(suffix="js")
   43 |+pure_posix_path.with_suffix(suffix=".js")
44 44 | 
45 45 | pure_windows_path.with_suffix(".")
46 46 | pure_windows_path.with_suffix("py")

PTH210.py:45:1: PTH210 Invalid suffix passed to `.with_suffix()`
   |
43 | pure_posix_path.with_suffix(suffix="js")
44 | 
45 | pure_windows_path.with_suffix(".")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
46 | pure_windows_path.with_suffix("py")
47 | pure_windows_path.with_suffix(r"s")
   |
   = help: Remove "." or extend to valid suffix

PTH210.py:46:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
45 | pure_windows_path.with_suffix(".")
46 | pure_windows_path.with_suffix("py")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
47 | pure_windows_path.with_suffix(r"s")
48 | pure_windows_path.with_suffix(u'' "json")
   |
   = help: Add a leading dot

ℹ Unsafe fix
43 43 | pure_posix_path.with_suffix(suffix="js")
44 44 | 
45 45 | pure_windows_path.with_suffix(".")
46    |-pure_windows_path.with_suffix("py")
   46 |+pure_windows_path.with_suffix(".py")
47 47 | pure_windows_path.with_suffix(r"s")
48 48 | pure_windows_path.with_suffix(u'' "json")
49 49 | pure_windows_path.with_suffix(suffix="js")

PTH210.py:47:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
45 | pure_windows_path.with_suffix(".")
46 | pure_windows_path.with_suffix("py")
47 | pure_windows_path.with_suffix(r"s")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
48 | pure_windows_path.with_suffix(u'' "json")
49 | pure_windows_path.with_suffix(suffix="js")
   |
   = help: Add a leading dot

ℹ Unsafe fix
44 44 | 
45 45 | pure_windows_path.with_suffix(".")
46 46 | pure_windows_path.with_suffix("py")
47    |-pure_windows_path.with_suffix(r"s")
   47 |+pure_windows_path.with_suffix(r".s")
48 48 | pure_windows_path.with_suffix(u'' "json")
49 49 | pure_windows_path.with_suffix(suffix="js")
50 50 | 

PTH210.py:48:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
46 | pure_windows_path.with_suffix("py")
47 | pure_windows_path.with_suffix(r"s")
48 | pure_windows_path.with_suffix(u'' "json")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
49 | pure_windows_path.with_suffix(suffix="js")
   |
   = help: Add a leading dot

ℹ Unsafe fix
45 45 | pure_windows_path.with_suffix(".")
46 46 | pure_windows_path.with_suffix("py")
47 47 | pure_windows_path.with_suffix(r"s")
48    |-pure_windows_path.with_suffix(u'' "json")
   48 |+pure_windows_path.with_suffix(u'.' "json")
49 49 | pure_windows_path.with_suffix(suffix="js")
50 50 | 
51 51 | windows_path.with_suffix(".")

PTH210.py:49:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
47 | pure_windows_path.with_suffix(r"s")
48 | pure_windows_path.with_suffix(u'' "json")
49 | pure_windows_path.with_suffix(suffix="js")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
50 | 
51 | windows_path.with_suffix(".")
   |
   = help: Add a leading dot

ℹ Unsafe fix
46 46 | pure_windows_path.with_suffix("py")
47 47 | pure_windows_path.with_suffix(r"s")
48 48 | pure_windows_path.with_suffix(u'' "json")
49    |-pure_windows_path.with_suffix(suffix="js")
   49 |+pure_windows_path.with_suffix(suffix=".js")
50 50 | 
51 51 | windows_path.with_suffix(".")
52 52 | windows_path.with_suffix("py")

PTH210.py:51:1: PTH210 Invalid suffix passed to `.with_suffix()`
   |
49 | pure_windows_path.with_suffix(suffix="js")
50 | 
51 | windows_path.with_suffix(".")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
52 | windows_path.with_suffix("py")
53 | windows_path.with_suffix(r"s")
   |
   = help: Remove "." or extend to valid suffix

PTH210.py:52:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
51 | windows_path.with_suffix(".")
52 | windows_path.with_suffix("py")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
53 | windows_path.with_suffix(r"s")
54 | windows_path.with_suffix(u'' "json")
   |
   = help: Add a leading dot

ℹ Unsafe fix
49 49 | pure_windows_path.with_suffix(suffix="js")
50 50 | 
51 51 | windows_path.with_suffix(".")
52    |-windows_path.with_suffix("py")
   52 |+windows_path.with_suffix(".py")
53 53 | windows_path.with_suffix(r"s")
54 54 | windows_path.with_suffix(u'' "json")
55 55 | windows_path.with_suffix(suffix="js")

PTH210.py:53:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
51 | windows_path.with_suffix(".")
52 | windows_path.with_suffix("py")
53 | windows_path.with_suffix(r"s")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
54 | windows_path.with_suffix(u'' "json")
55 | windows_path.with_suffix(suffix="js")
   |
   = help: Add a leading dot

ℹ Unsafe fix
50 50 | 
51 51 | windows_path.with_suffix(".")
52 52 | windows_path.with_suffix("py")
53    |-windows_path.with_suffix(r"s")
   53 |+windows_path.with_suffix(r".s")
54 54 | windows_path.with_suffix(u'' "json")
55 55 | windows_path.with_suffix(suffix="js")
56 56 | 

PTH210.py:54:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
52 | windows_path.with_suffix("py")
53 | windows_path.with_suffix(r"s")
54 | windows_path.with_suffix(u'' "json")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
55 | windows_path.with_suffix(suffix="js")
   |
   = help: Add a leading dot

ℹ Unsafe fix
51 51 | windows_path.with_suffix(".")
52 52 | windows_path.with_suffix("py")
53 53 | windows_path.with_suffix(r"s")
54    |-windows_path.with_suffix(u'' "json")
   54 |+windows_path.with_suffix(u'.' "json")
55 55 | windows_path.with_suffix(suffix="js")
56 56 | 
57 57 | 

PTH210.py:55:1: PTH210 [*] Dotless suffix passed to `.with_suffix()`
   |
53 | windows_path.with_suffix(r"s")
54 | windows_path.with_suffix(u'' "json")
55 | windows_path.with_suffix(suffix="js")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PTH210
   |
   = help: Add a leading dot

ℹ Unsafe fix
52 52 | windows_path.with_suffix("py")
53 53 | windows_path.with_suffix(r"s")
54 54 | windows_path.with_suffix(u'' "json")
55    |-windows_path.with_suffix(suffix="js")
   55 |+windows_path.with_suffix(suffix=".js")
56 56 | 
57 57 | 
58 58 | ### No errors
